azure-communication-ui/calling/src/main/java/com/azure/android/communication/ui/calling/presentation/fragment/setup/components/SetupControlBarViewModel.kt

```diff
@@ -87,41 +102,50 @@ internal class SetupControlBarViewModel(
         return audioPermissionState != PermissionStatus.DENIED
     }
 
-    val cameraIsEnabled: StateFlow<Boolean> get() = cameraIsEnabledStateFlow
-    val cameraIsVisible: StateFlow<Boolean> get() = cameraIsVisibleStateFlow
-    val micIsEnabled: StateFlow<Boolean> get() = micIsEnabledStateFlow
+    val cameraIsEnabled: StateFlow<Boolean> get() = isCameraButtonEnabledStateFlow
+    val cameraIsVisible: StateFlow<Boolean> get() = isCameraButtonVisibleStateFlow
+    val micIsEnabled: StateFlow<Boolean> get() = isMicButtonEnabledStateFlow
 
-    val deviceIsEnabled: StateFlow<Boolean> get() = deviceIsEnabledStateFlow
+    val audioDeviceButtonEnabled: StateFlow<Boolean> get() = audioDeviceButtonIsEnabledStateFlow
     val isVisibleState: StateFlow<Boolean> get() = visibleStateFlow
     val cameraState: StateFlow<CameraOperationalStatus> get() = cameraStateFlow
 
     val audioOperationalStatusStat: StateFlow<AudioOperationalStatus> get() = audioOperationalStatusStateFlow
     val audioDeviceSelectionStatusState: StateFlow<AudioState> get() = audioDeviceSelectionStatusStateFlow
 
-    fun turnCameraOn() {
+    fun turnCameraOn(context: Context) {
```

Comment:
as per above, this would look better if they were direct bindings for the button, and had the logic that is in the View be in the ViewModel instead for easier testing.